apps/collection/behaviours/confirm.js

```diff
@@ -32,6 +32,15 @@ module.exports = superclass => class Confirm extends superclass {
     const locals = super.locals(req, res);
     const reason = getReason(req);
     req.sessionModel.set(reason);
+    if (locals.route === 'confirm') {
+      locals.rows = locals.rows.map(row => {
+        if (req.sessionModel.get('nominated-fullname') && row.section === 'Details of the person who attempted to collect your BRP') {
+          row.section = `${req.sessionModel.get('nominated-fullname')}\n\nDetails of the person who attempted to collect your BRP`;
+          return row;
+        }
+        return row;
+      });
+    }
     return Object.assign({}, locals, {
```

Comment:
This is just a suggested alternative approach, you can ignore it but just a consideration to refactor the code

```
/* eslint-disable consistent-return, max-len */
'use strict';

const BRP_DETAILS_SECTION = 'Details of the person who attempted to collect your BRP';

function getReason(req) {
  const reasons = [
    'which-post-office',
    'under-age',
    'non-identity',
    'others-identity',
    'passport-family',
    'passport-lost',
    'no-brp',
    'other'
  ];

  const reason = {};
  const reasonType = req.form.values['reason-radio'];

  if (reasons.includes(reasonType)) {
    reason[reasonType] = req.sessionModel.get(reasonType);
  } else {
    // Handle invalid reason type (optional)
  }

  return reason;
}

module.exports = superclass => class Confirm extends superclass {
  locals(req, res) {
    const locals = super.locals(req, res);
    const reason = getReason(req);

    if (locals.route === 'confirm') {
      const modifiedRows = locals.rows.map(row => {
        if (req.sessionModel.get('nominated-fullname') && row.section === BRP_DETAILS_SECTION) {
          return {
            ...row,
            section: `${req.sessionModel.get('nominated-fullname')}\n\n${BRP_DETAILS_SECTION}`
          };
        }
        return row;
      });

      return Object.assign({}, locals, {
        rows: modifiedRows,
        reason
      });
    }

    return Object.assign({}, locals, {
      reason
    });
  }
};
```

Reply:
that makes sense and is more readable - I will refactor this 👍🏾

Reply:
Have refactored as suggested

Reply:
fab